src/include/duckdb/common/types/vector.hpp

```diff
@@ -127,7 +121,7 @@ class Vector {
 		return nullmask.all();
 	}
 
-protected:
+public:
```

Comment:
Rather than changing this to be public, please use a getter and setter to retrieve and change the value of these properties, i.e. add two functions:

```c++
inline VectorType Vector::GetVectorType() {
   return buffer->vector_type;
}
inline const LogicalType& Vector::GetType() {
     ...
}
```

(and perhaps also a setter for both of these, if required). 

As you may have noticed in this PR, making variables public directly is generally a bad idea, because it means internals are used all over the place. As a result changing these variables (either the name or their location) later on is a big pain. We made this mistake previously (hence why this PR needs to be this long!), let's not make it again :)

Reply:
Sure.